### PR TITLE
Added fix

### LINKED ISSUE #4620: [Bug] `stocks/options/eodchain --help` - error: the following arguments are required: -d/--date. + other items.
recreate with:

`/stocks/load aapl/options/exp 1/eodchain --help`

Screenshot 2023-03-29 at 12 16 31 AM

It is also not clear what happens after this command is used. There's no option to export from the function, and there is no way to see "what was loaded", or verify that it is the date selected.  What are the expected results of this function? It lives separately from the `load` command, the expectation of the user is that the entire option chain for the ticker will be loaded and presented as a single table.

Screenshot 2023-03-29 at 12 17 39 AM

### PR DESCRIPTION
**Pull request by the-praxs**
Fixed #4620 

- "--help" argument throws error when using with `eodchain` and `load` commands.
- [x] #4620 

# How has this been tested?
Tested by executing the terminal application in Python
and using the argument with the commands.

- [x] Make sure affected commands still run in terminal
- [x] Ensure the SDK still works
- [x] Check any related reports

# Checklist:

- [x] I have adhered to the GitFlow naming convention and my branch name is in the format of `feature/feature-name` or `hotfix/hotfix-name`.
- [x] Update [our documentation](https://openbb-finance.github.io/OpenBBTerminal/) following [these guidelines](https://github.com/OpenBB-finance/OpenBBTerminal/tree/main/website).  Update any user guides that are affected by the changes.
- [x] Update our tests following [these guidelines](https://github.com/OpenBB-finance/OpenBBTerminal/tree/main/tests).
- [x] Make sure you are following our [CONTRIBUTING guidelines](https://github.com/OpenBB-finance/OpenBBTerminal/blob/main/CONTRIBUTING.md).
- [x] If a feature was added make sure to add it to the corresponding [integration test script](https://github.com/OpenBB-finance/OpenBBTerminal/tree/develop/openbb_terminal/miscellaneous/integration_tests_scripts).

# Others
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.

### CODE REVIEW

**Comment by deeleeramone** [comment]
Same question as #4710, @jmaslek.

File: openbb_terminal/stocks/options/options_controller.py [hunk 1]
```diff
@@ -652,7 +652,7 @@ def call_load(self, other_args: List[str]):
             "--ticker",
             action="store",
             dest="ticker",
-            required="-h" not in other_args,
+            # required="-h" not in other_args,
```
Comment (hjoaquim) [inline openbb_terminal/stocks/options/options_controller.py:655]:
```suggestion
            required="-h" not in other_args and "--help" not in other_args,
```

File: openbb_terminal/stocks/options/options_controller.py [hunk 2]
```diff
@@ -1457,7 +1457,7 @@ def call_eodchain(self, other_args: List[str]):
             type=valid_date,
             dest="date",
             help="Date to get option chain for.  Format YYYY-MM-DD",
-            required="-h" not in other_args,
+            # required="-h" not in other_args,
```
Comment (hjoaquim) [inline openbb_terminal/stocks/options/options_controller.py:1460]:
```suggestion
            required="-h" not in other_args and "--help" not in other_args,
```

**Review by hjoaquim — CHANGES_REQUESTED** [review]
Hello @the-praxs, thanks a lot for contributing :rocket: 

In my comments, I've suggested a slightly different implementation where we actually check for the `-h` and `--help`.

What do you think?

If we don't, we can run into this problem if the user does not supply any args: 

image

**Comment by hjoaquim** [comment]
The above comment, also applies to #4710

**Comment by the-praxs** [comment]
@hjoaquim I also refactored some parts of the code, by using named expression to simplify assignment and conditional.

**Review by hjoaquim — APPROVED** [review]
Very nice additions, we don't actually have a lot of using walrus operators on the code base, so this is something we might want to look in the future :eyes: 

Well done @the-praxs

**Pull request merged**